scripts/unsupported/revert_sam/SamToUnmappedBams.wdl

```diff
@@ -0,0 +1,150 @@
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "1 GB"
+  }
+  output {
+    File output_bam = "${output_bam_name}"
+  }
+}
+
+task RevertSam {
+  File input_bam
+  File output_map
+  Int disk_size
+
+  command {
+    java -Xmx1000m -jar /usr/gitc/picard.jar \
+    RevertSam \
+    INPUT=${input_bam} \
+    OUTPUT_MAP=${output_map} \
+    OUTPUT_BY_READGROUP=true \
+    VALIDATION_STRINGENCY=LENIENT \
+    ATTRIBUTE_TO_CLEAR=FT \
+    ATTRIBUTE_TO_CLEAR=CO \
+    SORT_ORDER=coordinate
+  }
+  runtime {
+    docker: "broadinstitute/genomes-in-the-cloud:2.2.3-1469027018"
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "1200 MB"
+  }
+  output {
+    Array[File] unmapped_bams = glob("*.bam")
+  }
+}
+
+task SortSam {
+  File input_bam
+  String sorted_bam_name
+  Int disk_size
+
+  command {
+    java -Xmx3000m -jar /usr/gitc/picard.jar \
+    SortSam \
+    INPUT=${input_bam} \
+    OUTPUT=${sorted_bam_name} \
+    SORT_ORDER=queryname \
+    MAX_RECORDS_IN_RAM=1000000
+  }
+  runtime {
+    docker: "broadinstitute/genomes-in-the-cloud:2.2.3-1469027018"
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "3500 MB"
+    preemptible: 3
+  }
+  output {
+    File sorted_bam = "${sorted_bam_name}"
+  }
+}
+
+task ValidateSamFile {
+  File input_bam
+  String report_filename
+  Int disk_size
+
+  command {
+    java -Xmx3000m -jar /usr/gitc/picard.jar \
+      ValidateSamFile \
+      INPUT=${input_bam} \
+      OUTPUT=${report_filename} \
+      MODE=VERBOSE \
+      IS_BISULFITE_SEQUENCED=false 
+  }
+  runtime {
+    docker: "broadinstitute/genomes-in-the-cloud:2.2.3-1469027018"
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "3500 MB"
+    preemptible: 3
+  }
+  output {
+    File report = "${report_filename}"
+  }
+}
+
+workflow CramToUnmappedBams {
+  File? input_bam
+  File? input_cram
+  File? ref_fasta
+  File? ref_fasta_index
+  File output_map
+  String dir_pattern = "gs://.*/"
+  Int? cram_to_bam_disk_size
+  Int revert_sam_disk_size
+  Int sort_sam_disk_size
+  Int validate_sam_file_disk_size
+
+  String input_file_name = select_first([input_cram, input_bam])
+  String bam_from_cram_name = basename(input_file_name, ".cram") + ".bam"
+
+  if(defined(input_cram)) {
+    call CramToBam {
+      input:
+        input_cram = input_cram,
+        output_bam_name = bam_from_cram_name,
+        ref_fasta = ref_fasta,
+        ref_fasta_index = ref_fasta_index,
+        disk_size = cram_to_bam_disk_size
+    }
+  }
+
+  File input_data = select_first([CramToBam.output_bam, input_bam])
+
+  call RevertSam {
+    input:
+      input_bam = input_data,
+      output_map = output_map,
+      disk_size = revert_sam_disk_size
+  }
+
+  scatter (unmapped_bam in RevertSam.unmapped_bams) {
+    call SortSam {
+      input:
+        input_bam = unmapped_bam,
+        sorted_bam_name = sub(sub(unmapped_bam, dir_pattern, ""), ".coord.sorted.unmapped.bam$", "") + ".unmapped.bam",
```

Comment:
can probably take out the `sub(sub())` and use it in both places

Reply:
done